module/platformUserRoles/helper.js

```diff
@@ -185,4 +187,93 @@ module.exports = class platformUserRolesHelper {
 
     }
 
+    
+    static create(request, token) {
+
+        return new Promise(async (resolve, reject) => {
+            try {
+                
+                let body = request.body;
+
+                let response = await sunBirdService.createUser(body, request.userDetails.userToken);
+
+                if (response && response.responseCode == "OK") {
+
+                    
+                    let rolesId = [];
+                    let organisationsRoles= [];
+                    let plaformRoles = [];
+                    let allRoles =[];
+
+                    await Promise.all(request.body.roles.map(async function(roleInfo){
+
+                        let rolesDocument = await database.models.platformRolesExt.findOne({ _id:roleInfo.value },
+                            ).lean();
+
+                            if(rolesDocument){
+
+                                let roleObj = { 
+                                    roleId:rolesDocument._id,
+                                    code:rolesDocument.code,
+                                    name:rolesDocument.title
+                                }
+                                rolesId.push(roleObj);
+
+                                allRoles.push({ roleId:rolesDocument._id ,code:rolesDocument.code });
+                                if(rolesDocument && rolesDocument.platformRole &&
+                                     rolesDocument.platformRole==true ){
+
+                                   
+                                    plaformRoles.push(rolesDocument.code);
+                                }
+
+                            }
+                    }));
+                  
+                    let object ={
+                            "userId": response.result.userId,
+                            "organisationId": request.body.organisations.value,
+                            "roles": plaformRoles
+                        }
+                      
+                        organisationsRoles.push({ organisationId:request.body.organisations.value,roles:rolesId });
+                        let addUserToOrg = await sunBirdService.addUserToOrganisation(object,request.userDetails.userToken);
```

Comment:
@rakeshSgr - If the token is explicitly passed here in the function why use it again from request. Fix would be to not pass entire request here but only the body and token as 2 arguments